### PR TITLE
[src,scripts,egs] CNN+TDNN-F recipes

### PR DESCRIPTION
**Pull request by danpovey**

### CODE REVIEW

**Comment by GaofengCheng** [comment]
Nice improvement 
I will try this on switchboard first. 
After getting good results, I can try using blocksum layer.
Gaofeng

**Comment by danpovey** [comment]
Thanks a lot, @kangshiyin; I merged your PR and merged the latest master into this PR.

**Comment by danpovey** [comment]
I added a WSJ example.  It's a little better, but not a lot.  I will try to tune it more; right now I'm trying taking out two TDNN-F layers but I don't have much hope that it will help, based on my experience in mini-librispeech.  @GaofengCheng, maybe you could also try it with slightly larger l2-regularize values, since it seems to be overfitting more-- if you have a WSJ setup.

File: egs/wsj/s5/local/chain/tuning/run_cnn_tdnn_1b17.sh [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+  conv-relu-batchnorm-layer name=cnn1 $cnn_opts height-in=40 height-out=40 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=48 learning-rate-factor=0.333 max-change=0.25
+  conv-relu-batchnorm-layer name=cnn2 $cnn_opts height-in=40 height-out=40 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=48
+  conv-relu-batchnorm-layer name=cnn3 $cnn_opts height-in=40 height-out=20 height-subsample-out=2 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=64
+  conv-relu-batchnorm-layer name=cnn4 $cnn_opts height-in=20 height-out=20 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=64
+  conv-relu-batchnorm-layer name=cnn5 $cnn_opts height-in=20 height-out=10 height-subsample-out=2 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=64
+  conv-relu-batchnorm-layer name=cnn6 $cnn_opts height-in=10 height-out=5 height-subsample-out=2 time-offsets=-1,0,1 height-offsets=-1,0,1 num-filters-out=128
+
+  # the first TDNN-F layer has no bypass (since dims don't match), and a larger bottleneck so the
+  # information bottleneck doesn't become a problem.
+  tdnnf-layer name=tdnnf7 $tdnnf_first_opts dim=1024 bottleneck-dim=256 time-stride=0
```
Comment (danpovey) [inline egs/wsj/s5/local/chain/tuning/run_cnn_tdnn_1b17.sh]: The 256 here was a mistake; I intended for it to be the same as the final-layer bottleneck which it seems is 192 in the WSJ setup.

**Comment by danpovey** [comment]
@xiaohui-zhang, do you have time to help with this PR?  I just need the scripts to be renamed to the next consecutive letter (no number), the references to the script/affixes inside the scripts themselves to be corrected, and the comparisons with non-checked-in scripts to be removed from the comments in the headers.
I'm super busy with something right now so I don't have time to do this.

I was originally hoping @GaofengCheng would be able to get CNN+TDNN-F to help (vs. TDNN-F) on larger setups,as our grid was busy, but I think he was not able to, so far.  I think we can just check this in now, rather than wait.

**Comment by xiaohui-zhang** [comment]
no prob!

**Comment by xiaohui-zhang** [comment]
done

**Comment by danpovey** [comment]
Closing since merged via #2643

**Pull request closed without merging**